src/Mod/TechDraw/Gui/QGIViewBalloon.h

```diff
@@ -163,7 +163,7 @@ public Q_SLOTS:
     void updateBalloon(bool obtuse = false);
 
 protected:
-    void draw() override;
```

Comment:
I don't know the internal details here, but by not providing the virtual version of `draw()`, you run the risk of breaking anything that was trying to use a base class pointer to access the subclass's `draw()` functions. It might be a good idea to implement the `void draw() override` to have it call your newly-revised function.

Reply:
I lack knowledge here, so please correct me: There is a base draw() void. Before my patch this one was overriden to use the draw() defined for Balloon. Now I use a different void. I could have given it a completely different name but was lazy. Since I declared it in the private: section, I thought all calls from within QGIViewBalloon.cpp go now to the new draw(bool) void, no?

I can and will now change it as you recommended but would like to understand why this is better.

Reply:
My concern was that this class used to provide an overridden `draw()` function, but now no longer does. I didn't look to see if that function did anything, I just saw its removal and was concerned. If you've checked and that function was not needed, then no need to change anything. You are correct that your new function will not be called by a base class call to `draw()`, since your function is non-virtual (and has a different signature besides).

Reply:
thanks for the reply. However, I followed your suggestion and kept now the draw() overriden and use it as redirect.